alarm_active_table.hpp

```diff
@@ -0,0 +1,128 @@
+/**
+* Project Clearwater - IMS in the Cloud
+* Copyright (C) 2014 Metaswitch Networks Ltd
+*
+* This program is free software: you can redistribute it and/or modify it
+* under the terms of the GNU General Public License as published by the
+* Free Software Foundation, either version 3 of the License, or (at your
+* option) any later version, along with the "Special Exception" for use of
+* the program along with SSL, set forth below. This program is distributed
+* in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+* without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+* A PARTICULAR PURPOSE. See the GNU General Public License for more
+* details. You should have received a copy of the GNU General Public
+* License along with this program. If not, see
+* <http://www.gnu.org/licenses/>.
+*
+* The author can be reached by email at clearwater@metaswitch.com or by
+* post at Metaswitch Networks Ltd, 100 Church St, Enfield EN2 6BQ, UK
+*
+* Special Exception
+* Metaswitch Networks Ltd grants you permission to copy, modify,
+* propagate, and distribute a work formed by combining OpenSSL with The
+* Software, or a work derivative of such a combination, even if such
+* copying, modification, propagation, or distribution would otherwise
+* violate the terms of the GPL. You must comply with the GPL in all
+* respects for all of the code used other than OpenSSL.
+* "OpenSSL" means OpenSSL toolkit software distributed by the OpenSSL
+* Project and licensed under the OpenSSL Licenses, or a work based on such
+* software and licensed under the OpenSSL Licenses.
+* "OpenSSL Licenses" means the OpenSSL License and Original SSLeay License
+* under which the OpenSSL Project distributes the OpenSSL toolkit software,
+* as those licenses appear in the file LICENSE-OPENSSL.
+*/
+
+/*
+ *  Note: this file originally auto-generated by mib2c using
+ *         : mib2c.array-user.conf 15997 2007-03-25 22:28:35Z dts12 $
+ *
+ *  $Id:$
+ */
+
+#ifndef ALARM_ACTIVE_TABLE_HPP
+#define ALARM_ACTIVE_TABLE_HPP
+
+#include <alarm_table_defs.hpp>
+#include "alarm_trap_sender.hpp"
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <net-snmp/net-snmp-config.h>
+#include <net-snmp/net-snmp-includes.h>
+#include <net-snmp/agent/net-snmp-agent-includes.h>
+
+#include <net-snmp/library/snmp_assert.h>
+    
+#include <net-snmp/net-snmp-config.h>
+#include <net-snmp/library/container.h>
+#include <net-snmp/agent/table_array.h>
+
+#include <string>
+using namespace std;
+
+typedef struct alarmActiveTable_context_s {
+    netsnmp_index _index; /** THIS MUST BE FIRST!!! */
+
+    AlarmTableDef* _alarm_table_def;
+
+} alarmActiveTable_context;
+
+/*************************************************************
+ *
+ *  function declarations
```

Comment:
I'm not sure this comment is necessary - it's pretty obvious that these are function declarations.